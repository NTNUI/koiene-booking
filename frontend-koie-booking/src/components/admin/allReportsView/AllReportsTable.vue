<template>
  <v-data-table
    style="width: 100%"
    :headers="headers"
    :items="allReports"
    :items-per-page="24"
    :hide-default-footer="true"
    loading-text="Henter inn rapporter"
    no-data-text="Det skjedde noe feil da vi prøvde å hente rapporter"
  >
    <template slot="item" slot-scope="row">
      <ReportRow :report="row.item" />
    </template>
  </v-data-table>
</template>

<script lang="ts">
import Vue from 'vue';
import { reportTableHeaders } from '@/components/admin/allReportsView/ReportTableHeaders';
import getReportData from '../../../../tests/unit/__mocks__/reports';
import ReportRow from '@/components/admin/allReportsView/ReportRow.vue';
import AdminReport from '@/types/admin/APIAdminReport';

export default Vue.extend({
  name: 'AllReportsTable',
  components: { ReportRow },
  data() {
    return {
      headers: reportTableHeaders,
    };
  },
  computed: {
    allReports(): Array<AdminReport> {
      return this.$store.getters['adminReports/getReportArray'];
    },
  },
  mounted() {
    this.$store.dispatch('adminReports/MOUNT_REPORTS');
  },
  methods: {},
});
</script>

<style scoped></style>
