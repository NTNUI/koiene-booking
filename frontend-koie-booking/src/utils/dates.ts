import dayjs from 'dayjs';
require('dayjs/locale/nb');
dayjs.locale('nb');

export function getDateString(baseDate?: string, addDays?: number): string {
  let resultString: string;
  if (baseDate && addDays) {
    resultString = new Date(new Date(baseDate).getTime() + addDays * 24 * 60 * 60 * 1000).toISOString().substr(0, 10);
  } else if (baseDate) {
    resultString = new Date(baseDate).toISOString().substr(0, 10);
  } else if (addDays) {
    resultString = new Date(new Date().getTime() + addDays * 24 * 60 * 60 * 1000).toISOString().substr(0, 10);
  } else {
    resultString = new Date().toISOString().substr(0, 10);
  }
  return resultString;
}

export function getDate(baseDate?: string, addDays?: number): Date {
  let resultDate: Date;
  if (baseDate && addDays) {
    resultDate = new Date(new Date(baseDate).getTime() + addDays * 24 * 60 * 60 * 1000);
  } else if (baseDate) {
    resultDate = new Date(baseDate);
  } else if (addDays) {
    resultDate = new Date(new Date().getTime() + addDays * 24 * 60 * 60 * 1000);
  } else {
    resultDate = new Date();
  }
  return resultDate;
}

export function addToDate(startDate: string, howMany: number, timeUnit: dayjs.OpUnitType): string {
  const date = dayjs(startDate + 'T00:00:00.000Z');
  return date.add(howMany, timeUnit).format('YYYY-MM-DD');
}

export function formatDate(dateISO: string, formatString: string): string {
  const date = dayjs(dateISO + 'T00:00:00.000Z');
  return date.format(formatString);
}
