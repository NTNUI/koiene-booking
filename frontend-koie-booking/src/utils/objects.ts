export const clone = (object: any) => Object.assign({}, object);

export const renameKey = (object: any, key: string, newKey: string): any => {
  const value = object[key];
  delete object[key];
  object[newKey] = value;
  return object;
};

export const sortObjectByKey = (unordered: any) => {
  let ordered: any;
  ordered = {};
  Object.keys(unordered)
    .sort()
    .forEach(function(key) {
      ordered[key] = unordered[key];
    });
  return ordered;
};
