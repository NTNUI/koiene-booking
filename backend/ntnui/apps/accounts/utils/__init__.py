from accounts.utils.mail_utils import send_gsuite_confirmation_mail
