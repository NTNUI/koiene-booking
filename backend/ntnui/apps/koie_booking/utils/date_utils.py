from django.utils.timezone import timedelta


def get_daterange(start_date, end_date):
    for n in range(int((end_date - start_date).days)):
        yield start_date + timedelta(n)
