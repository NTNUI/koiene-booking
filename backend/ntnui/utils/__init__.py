from ntnui.utils.pagination import PageSize7, PageSize20, PageSize100
