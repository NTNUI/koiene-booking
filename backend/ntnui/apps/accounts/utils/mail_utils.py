from django.core.mail import send_mail
from django.template.loader import render_to_string

from ntnui.enums import MembershipType


def new_membership(membership):
    """ Sends confirmation email to users and guests who signs up for events and sub-events. """

    # Mail header.
    sender = "noreply@mg.ntnui.no"
    receiver = [membership.member.email]
    subject = "Welcome"

    context = {"user": membership.member, "group": membership.group}

    # Mail body.
    plain_message = render_to_string("emails/new_membership.txt", context)
    html_message = render_to_string("emails/new_membership.html", context)

    send_mail(subject, plain_message, sender, receiver, html_message=html_message)


def send_gsuite_confirmation_mail(user, contact_email, password):
    # Mail header.
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = "Opprettelse av ny NTNUI-bruker"

    context = {
        "first_name": user.first_name,
        "email": user.email,
        "contact_email": contact_email,
        "password": password,
    }

    # Mail body.
    plain_message = render_to_string("emails/new_gsuite_account.txt", context)
    html_message = render_to_string("emails/new_gsuite_account.html", context)

    send_mail(subject, plain_message, sender, receiver, html_message=html_message)


def send_welcome_mail_to_leader(user, group):
    """ Send a mail to a new leader """
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = f"Velkommen som ny leder i {group}"

    context = {
        "first_name": user.first_name,
        "email": user.get_contact_email(),
        "group": group.name,
    }

    plain_message = render_to_string(
        template_name="emails/new_leader_membership.txt", context=context
    )
    html_message = render_to_string(
        template_name="emails/new_leader_membership.html", context=context
    )

    send_mail(
        subject=subject,
        message=plain_message,
        from_email=sender,
        recipient_list=receiver,
        html_message=html_message,
    )


def send_welcome_mail_to_cashier(user, group):
    """Send a mail to the new cashier"""
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = f"Velkommen som ny kasserer i {group}"

    context = {
        "first_name": user.first_name,
        "email": user.get_contact_email(),
        "group": group.name,
    }

    plain_message = render_to_string(
        template_name="emails/new_cashier_membership.txt", context=context
    )
    html_message = render_to_string(
        template_name="emails/new_cashier_membership.html", context=context
    )

    send_mail(
        subject=subject,
        message=plain_message,
        from_email=sender,
        recipient_list=receiver,
        html_message=html_message,
    )


def send_welcome_mail_to_board_member(user, group):
    """Send a mail to the new cashier"""
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = f"Velkommen som ny tillitsvalgt i {group}"

    context = {
        "first_name": user.first_name,
        "email": user.get_contact_email(),
        "group": group.name,
    }

    plain_message = render_to_string(
        template_name="emails/new_board_membership.txt", context=context
    )
    html_message = render_to_string(
        template_name="emails/new_board_membership.html", context=context
    )

    send_mail(
        subject=subject,
        message=plain_message,
        from_email=sender,
        recipient_list=receiver,
        html_message=html_message,
    )


def send_welcome_mail(membership):
    """ Delegate welcome mail to different board members """
    send_welcome_mail_to_board_member(user=membership.member, group=membership.group)
    if membership.type == MembershipType.leader:
        send_welcome_mail_to_leader(user=membership.member, group=membership.group)

    if membership.type == MembershipType.cashier:
        send_welcome_mail_to_cashier(user=membership.member, group=membership.group)
