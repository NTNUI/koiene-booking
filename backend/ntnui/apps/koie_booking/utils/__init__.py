from koie_booking.utils.mail_utils import send_confirmation_mail, send_koie_information_mail
from koie_booking.utils.date_utils import get_daterange
