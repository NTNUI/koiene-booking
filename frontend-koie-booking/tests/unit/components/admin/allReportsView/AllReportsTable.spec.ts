import Vue from 'vue';
import Vuetify from 'vuetify';
import { createWrapper } from '../../../utils';
import { Wrapper } from '@vue/test-utils';

import AllReportsTable from '@/components/admin/allReportsView/AllReportsTable.vue';
Vue.use(Vuetify);

describe('Component AllReportsTable', () => {
  let wrapper: Wrapper<any>;

  beforeEach(() => {
    const wrapperOptions = {
      sync: false,
    };
    wrapper = createWrapper(AllReportsTable, wrapperOptions);
  });

  it('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
