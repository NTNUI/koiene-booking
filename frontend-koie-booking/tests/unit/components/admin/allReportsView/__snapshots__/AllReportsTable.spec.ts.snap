// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component AllReportsTable Matches snapshot 1`] = `
<div class="v-data-table theme--light" style="width: 100%;">
  <div class="v-data-table__wrapper">
    <table>
      <colgroup>
        <col class="">
        <col class="">
        <col class="">
        <col class="">
        <col class="">
        <col class="">
        <col class="">
        <col class="">
        <col class="">
        <col class="">
        <col class="">
        <col class="">
        <col class="">
      </colgroup>
      <thead class="v-data-table-header">
        <tr>
          <th role="columnheader" scope="col" aria-label="Tidsmerke: Not sorted. Activate to sort ascending." aria-sort="none" class="text-center sortable" style="width: 7%; min-width: 7%;"><span>Tidsmerke</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
          <th role="columnheader" scope="col" aria-label="Koie: Not sorted. Activate to sort ascending." aria-sort="none" class="text-center sortable" style="width: 9%; min-width: 9%;"><span>Koie</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
          <th role="columnheader" scope="col" aria-label="Turdato: Not sorted. Activate to sort ascending." aria-sort="none" class="text-center sortable" style="width: 7%; min-width: 7%;"><span>Turdato</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
          <th role="columnheader" scope="col" aria-label="Gassbestand" class="text-center" style="width: 6%; min-width: 6%;"><span>Gassbestand</span></th>
          <th role="columnheader" scope="col" aria-label="Vedstokker" class="text-center" style="width: 7%; min-width: 7%;"><span>Vedstokker</span></th>
          <th role="columnheader" scope="col" aria-label="Opphuggd ved" class="text-center" style="width: 8%; min-width: 8%;"><span>Opphuggd ved</span></th>
          <th role="columnheader" scope="col" aria-label="Båt" class="text-center" style="width: 9%; min-width: 9%;"><span>Båt</span></th>
          <th role="columnheader" scope="col" aria-label="Kano" class="text-center" style="width: 9%; min-width: 9%;"><span>Kano</span></th>
          <th role="columnheader" scope="col" aria-label="Redningsvest" class="text-center" style="width: 6%; min-width: 6%;"><span>Redningsvest</span></th>
          <th role="columnheader" scope="col" aria-label="Røykvarsler" class="text-center" style="width: 5%; min-width: 5%;"><span>Røykvarsler</span></th>
          <th role="columnheader" scope="col" aria-label="Utstyr" class="text-center" style="width: 150px; min-width: 150px;"><span>Utstyr</span></th>
          <th role="columnheader" scope="col" aria-label="Andre utstyrsfeil" class="text-center" style="width: 9%; min-width: 9%;"><span>Andre utstyrsfeil</span></th>
          <th role="columnheader" scope="col" aria-label="Tilbakemelding" class="text-center" style="width: 8%; min-width: 8%;"><span>Tilbakemelding</span></th>
        </tr>
      </thead>
      <thead>
        <tr class="v-data-table__progress">
          <th colspan="13" class="column">
            <div role="progressbar" aria-valuemin="0" aria-valuemax="100" class="v-progress-linear theme--light" style="height: 2px;">
              <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 0%; width: 100%;"></div>
              <div class="v-progress-linear__buffer"></div>
              <div class="v-progress-linear__indeterminate v-progress-linear__indeterminate--active">
                <div class="v-progress-linear__indeterminate long primary"></div>
                <div class="v-progress-linear__indeterminate short primary"></div>
              </div>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr class="v-data-table__empty-wrapper">
          <td colspan="13">Henter inn rapporter</td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
